kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/FloatingPointParser.kt

```diff
@@ -201,66 +304,32 @@ object FloatingPointParser {
     }
 
     /*
-     * Assumes the string is trimmed.
+     * Assumes the string is trimmed. This method is used for both Double and Float "named" numbers.
+     * This method was needed to unify the common logic for Double and Float processing.
+     * "Inifinity" and "NaN" string values will be covered by this method.
      */
-    private fun parseDoubleName(namedDouble: String, length: Int): Double {
+    private fun <T> parseNamed(namedFloat: String, length: Int,
+                               negativeInf: T, positiveInf: T, nan: T): T {
         // Valid strings are only +Nan, NaN, -Nan, +Infinity, Infinity, -Infinity.
         if (length != 3 && length != 4 && length != 8 && length != 9) {
             throw NumberFormatException()
         }
 
         var negative = false
-        var cmpstart = 0
-        when (namedDouble[0]) {
+        val cmpstart = when (namedFloat[0]) {
             '-' -> {
                 negative = true
-                cmpstart = 1
+                1
             }
-            '+' -> cmpstart = 1
-        }
-
-        if (namedDouble.regionMatches(cmpstart, "Infinity", 0, 8, ignoreCase = false)) {
-            return if (negative)
-                Double.NEGATIVE_INFINITY
-            else
-                Double.POSITIVE_INFINITY
+            '+' -> 1
+            else -> 0
         }
 
-        if (namedDouble.regionMatches(cmpstart, "NaN", 0, 3, ignoreCase = false)) {
-            return Double.NaN
+        return when (namedFloat.subSequence(cmpstart, length)) {
```

Comment:
This logic will fix minor issue with incorrect "Inifinity" and "NaN" input values. 
Now we will check exact match instead of checking first 4/9 symbols.

No affect on performace, as the string is already limited to 9 symbols.